content/en/docs/concepts/cluster-administration/networking.md

```diff
@@ -21,56 +22,37 @@ default.  There are 4 distinct networking problems to solve:
-are on the same machine (and thus the same virtual bridge).  Containers on
-different machines can not reach each other - in fact they may end up with the
-exact same network ranges and IP addresses.
-
-In order for Docker containers to communicate across nodes, there must
-be allocated ports on the machine’s own IP address, which are then
-forwarded or proxied to the containers. This obviously means that
-containers must either coordinate which ports they use very carefully
-or ports must be allocated dynamically.
-
-## Kubernetes model
-
-Coordinating ports across multiple developers is very difficult to do at
-scale and exposes users to cluster-level issues outside of their control.
-Dynamic port allocation brings a lot of complications to the system - every
-application has to take ports as flags, the API servers have to know how to
-insert dynamic port numbers into configuration blocks, services have to know
-how to find each other, etc.  Rather than deal with this, Kubernetes takes a
-different approach.
+Kubernetes is all about sharing machines between applications.  Typically,
+sharing machines requires ensuring that two applications on the same do not try
+to use the same ports.  Coordinating ports across multiple developers is very
+difficult to do at scale and exposes users to cluster-level issues outside of
+their control.  Dynamic port allocation brings a lot of complications to the
+system - every application has to take ports as flags, the API servers have to
+know how to insert dynamic port numbers into configuration blocks, services
+have to know how to find each other, etc.  Rather than deal with this,
+Kubernetes takes a different approach.
+
+## The Kubernetes network model
+
+Every `Pod` gets its own IP address. This means you do not need to explicitly
+create links between `Pods` and you almost never need to deal with mapping
+container ports to host ports.  This creates a clean, backwards-compatible
+model where `Pods` can be treated much like VMs or physical hosts from the
+perspectives of port allocation, naming, service discovery, load balancing,
+application configuration, and migration.
 
 Kubernetes imposes the following fundamental requirements on any networking
 implementation (barring any intentional network segmentation policies):
 
-   * all containers can communicate with all other containers without NAT
-   * all nodes can communicate with all containers (and vice-versa) without NAT
-   * the IP that a container sees itself as is the same IP that others see it as
+   * pods on any node can communicate with all pods on all nodes without NAT
+   * agents on a node can communicate with all pods on that node
+   * the IP that a pod sees itself as is the same IP that others see it as
+
+In addition to these, for those platforms that support `Pods` running in the host
+network (e.g. Linux supports this):
+
+   * pods in the host network of any node can communicate with all pods on all
```

Comment:
no, this was always intended, and was implied by the first rule (all pods <-> all pods).  Otherwise some pods are not able to reach other pods.

I think it is in e2e, too.

As much as I dislike it, some apps run hostNetwork for performance reasons.  Removing this constraint would potentially break them.